src/notifier/keep_alive_monitor.py

```diff
@@ -24,21 +27,45 @@ class KeepAliveMonitor:
     receiving keep-alive ping events and can notify the user.
     """
 
-    def __init__(self, config: Optional[dict] = None, thresholds: Optional[dict] = None):
+    def __init__(self, config: ConfigView):
         self._notify_manager = None
+        self.config = config
 
-        self._last_keep_alive = {EventService.HARVESTER: datetime.now()}
-        self._last_keep_alive_threshold_seconds = thresholds or {EventService.HARVESTER: 300}
+        # These will be set by set_services() once known
+        self._last_keep_alive: Dict[EventService, datetime] = {}
+        self._last_keep_alive_threshold_seconds: Dict[EventService, int] = {}
```

Comment:
As discussed in previous comment, `set_services` would ideally be called directly below these lines so we know for sure they’re being initialized.